app/controllers/api/physical_servers_controller.rb

```diff
@@ -41,37 +41,77 @@ def restart_mgmt_controller_resource(type, id, _data)
     end
 
     def refresh_resource(type, id, _data = nil)
-      raise BadRequestError, "Must specify an id for refreshing a #{type} resource" unless id
+      refresh_physical_server("refresh_ems", type, id)
```

Comment:
I don't believe I see `refresh_physical_server` anywhere else? I am not sure I see the need for a separate method. I also agree that perhaps this could be a separate PR/commit to make it more clear